src/utils/classNameUsageCheckInjector.js

```diff
@@ -13,13 +15,18 @@ export default (target: Object) => {
       targetCDM.call(this);
     }
 
+    const componentName = getComponentName(this.props.forwardedClass.target);
```

Comment:
I don't think this is going to be high fidelity enough because `getComponentName` can resolve to "Component" if `displayName` isn't set and warnings won't be emitted.

What we can do instead is add `this.type` (which is the unique constructor class/function I think?) to the Set and that way it'll still track each new component individually in case they share the same name.

Reply:
@probablyup As I mentioned above, React has the same solution (and has "Component" too in some cases). I have tried `this.type` but it was `undefined` in tests.

Reply:
How about `this.constructor`?

Reply:
The React code is fine for putting a name in a warning, but not good enough for tracking which components have received warnings.

Reply:
@probablyup Ok, I have found how to obtain type: `this. _reactInternalFiber.type`. Are you admit to use it?

Reply:
`this.constructor` doesn't work? I tried it out in a sandbox...